### PR TITLE
Basic UI for viewing events

### PR DESCRIPTION
**Pull request by sdunster**
* Uses client-mocked data for now but should be pretty easy to swap in with stuff from server.
* This view is designed to be view-only for users who want to see what is happening with an event but aren't necessarily responding to or otherwise interacting with it. I suspect we will build a seperate responder interface that allows updates of your responding status.

### CODE REVIEW

**Comment by sdunster** [comment]
This PR is based off of #71 and #74 so I suggest you only review the latest commit: 9d089ba

Tested only on iOS:
screen shot 2017-11-26 at 22 24 09
screen shot 2017-11-26 at 22 24 11

Work still to be done:
* Replace MAP placeholder with actual map. Possibly this: https://github.com/airbnb/react-native-maps
* Hook it up to the actual API rather than use mocks. We might need to re-jigger the fields a bit since the API is pretty bare atm.
* Decide what metadata we actually want to show.
* Set up subscriptions so this updates in real time.

**Comment by sdunster** [comment]
#71 and #74 landed - rebased onto master.

File: client/src/screens/event-detail.screen.js [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+    <View style={styles.respondContainer}>
+      <Icon name="user" size={24} color={color} />
+      <View style={styles.respondTextContainer}>
+        <Text style={styles.respondName} numberOfLines={1}>{user.name}</Text>
+        <Text style={styles.respondStatus} numberOfLines={1}>{status} - {detail}</Text>
+      </View>
+    </View>
+  );
+};
+EventResponse.propTypes = {
```
Comment (OSPFNeighbour) [inline client/src/screens/event-detail.screen.js]: probably all isRequired

File: client/src/screens/event-detail.screen.js [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+          renderItem={this.renderItem}
+          onRefresh={this.onRefresh}
+          extraData={event} // redraw if this changes
+          refreshing={networkStatus === 4}
+        />
+      </View>
+    );
+  }
+}
+EventDetail.propTypes = {
```
Comment (OSPFNeighbour) [inline client/src/screens/event-detail.screen.js]: again I would suggest some of these are ```isRequired``` to catch malformed data

**Comment by OSPFNeighbour** [comment]
Seems about right.

**Pull request merged**